Realm/RLMSchema.mm

```diff
@@ -163,15 +162,24 @@ + (instancetype)dynamicSchemaFromRealm:(RLMRealm *)realm {
     return schema;
 }
 
-NSUInteger RLMRealmSchemaVersion(RLMRealm *realm) {
+NSUInteger RLMRealmSchemaVersion(RLMRealm *realm)
+{
     realm::TableRef table = realm.group->get_table(c_metadataTableName);
     if (!table || table->get_column_count() == 0) {
         return RLMNotVersioned;
     }
-    return NSUInteger(table->get_int(c_versionColumnIndex, 0));
+
+    uint64_t version = table->get_int(c_versionColumnIndex, 0);
+
+    const uint64_t old64BitRLMNotVersioned = std::numeric_limits<uint64_t>::max();
+    if (version == old64BitRLMNotVersioned) {
```

Comment:
This can't actually happen (barring a *very* strange bug, the schema version is always set to an actual version before the initial transaction is committed).